File: 프로그래머스/2/178870. 연속된 부분 수열의 합/연속된 부분 수열의 합.py
def solution(sequence, k):
    n = len(sequence)
    left = total = 0
    result = []
    min_len = float('inf')

    for right in range(n):
        total += sequence[right]

        while total > k:
            total -= sequence[left]
            left += 1

        if total == k:
            if right - left < min_len:
                min_len = right - left
                result = [left, right]

    return result

File: 프로그래머스/2/178870. 연속된 부분 수열의 합/연속된 부분 수열의 합.js
function solution(sequence, k) {
    const N = sequence.length;
    let left = 0, total = 0;
    let result = [];
    let minLength = Infinity;
    
    for(let right = 0 ; right < N ; right++){
        total += sequence[right];
        
        while(total > k){
            total -= sequence[left]
            left += 1
        }
        
        if(total === k){
            if(right - left < minLength){
                minLength = right - left;
                result = [left, right];
            }
        }
    }
    
    return result;
}